test/unit/features/segment_deletion_config_test.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Features::SegmentDeletionConfigTest < ActiveSupport::TestCase
+  def setup
+    @valid_config = {'email' => 'email@example.com', 'password' => 'example-password'}
+  end
+
+  attr_reader :valid_config
+
+  test 'marks as invalid if any value is missing' do
+    valid_config.each_key do |k|
+      config = valid_config.dup
+      config.delete(k)
+      refute segment_deletion_config(config).enabled?
+    end
+  end
+
+  test 'enabled for the right config in valid environments' do
+    valid_config.each { |key, value| assert_equal value, segment_deletion_config.config.public_send(key) }
+
+    Features::SegmentDeletionConfig.stubs(valid_environment?: false)
+    refute segment_deletion_config.enabled?
+
+    Features::SegmentDeletionConfig.stubs(valid_environment?: true)
+    assert segment_deletion_config.enabled?
+  end
+
+  test 'only production and test are valid environments' do
+    enabled_environments = %w[production test]
+    disabled_environments = available_environments - enabled_environments
+
+    enabled_environments.each  { |env| assert Features::SegmentDeletionConfig.valid_environment?(environment: env) }
+    disabled_environments.each { |env| refute Features::SegmentDeletionConfig.valid_environment?(environment: env) }
+  end
+
+  private
+
+  def available_environments
+    @available_environments ||= Dir.glob('./config/environments/*.rb').map { |filename| File.basename(filename, '.rb') }
```

Comment:
You could also get that from `ActiveRecord::Base.configurations.keys`.

Reply:
Woooooooow that's cool! I hope that actually works too 👍 I will try 👍

Reply:
@guicassolato unfortunately it doesn't contain the same for me 😭 it contains:
```ruby
["default", "production", "development", "test"]
```

And we need:
```ruby
["preview", "production", "development", "test"]
```

Reply:
Yeah... We would be mixing up unrelated namespaces anyway. In the end, `ActiveRecord::Base` has nothing to do with `SegmentDeletionConfig`.